### PR TITLE
RDCC-78: Retrieve professional user by email

### PR DESCRIPTION
**Pull request by TabbyC**
https://tools.hmcts.net/jira/browse/RDCC-78

- Add a new endpoint: GET /v1/organisation/users?email={email}
  - If the email address in the request URL is found in the database, the corresponding user is returned, else a 404.
- Bring PIT test coverage to 100%
- Fix bug with creating organisation with pba account where user was not linked to the account

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @param email The email address to search for
+     * @return The user with the matching email address
+     * @throws HTTPException with the status set to 404 if the email address was not
+     *                       found
+     */
+    public ProfessionalUser findProfessionalUserByEmailAddress(String email) {
+        ProfessionalUser user = new ProfessionalUser();
+        user.setEmailAddress(email);
+        return professionalUserRepository.findOne(Example.of(user)).orElseThrow(() -> {
```
Comment (t80) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserService.java]: Is this better than findByEmailAddress? seems messier have to create an object to compare

File: src/main/java/uk/gov/hmcts/reform/professionalapi/infrastructure/controllers/SearchController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.hmcts.reform.professionalapi.infrastructure.controllers.response.ProfessionalUserResponse;
+
+@RequestMapping(
+        path = "/search",
+        produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+@RestController
+public class SearchController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OrganisationController.class);
+
```
Comment (t80) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/infrastructure/controllers/SearchController.java]: We've been using @SL4J to inject the logger

File: src/test/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                .thenReturn(Optional.ofNullable(professionalUser));
+
+        ProfessionalUser user = professionalUserService.findProfessionalUserByEmailAddress("some-email");
+        assertEquals(professionalUser.getFirstName(), user.getFirstName());
+        assertEquals(professionalUser.getLastName(), user.getLastName());
+        assertEquals(professionalUser.getEmailAddress(), user.getEmailAddress());
+    }
+    
+    @Test (expected = HTTPException.class)
+    public void retrieveUserByEmailNotFound() {
```
Comment (t80) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserServiceTest.java]: assertJ has nice syntax for asserting that exceptions are thrown, same end result however

File: src/main/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserService.java (hunk 1 above)
Comment (TabbyC) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/domain/service/ProfessionalUserService.java]: done

File: src/main/java/uk/gov/hmcts/reform/professionalapi/infrastructure/controllers/SearchController.java (hunk 2 above)
Comment (TabbyC) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/infrastructure/controllers/SearchController.java]: Not even using the logger in that class. Removed

**Pull request merged**